File: data/src/main/java/com/github/gmetal/data/entity/response/BasePagedResponseEntity.java
package com.github.gmetal.data.entity.response;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;
import com.github.gmetal.data.entity.BaseMoviesDbEntity;

import java.util.List;

@JsonObject
public class BasePagedResponseEntity<T> extends BaseMoviesDbEntity<T> {

    @JsonField(name = "page")
    public int mPage;

    @JsonField(name = "total_results")
    public int mTotalResults;

    @JsonField(name = "total_pages")
    public int mTotalPages;

    public List<T> getResultObjects() {

        return null;
    }
}

File: data/src/main/java/com/github/gmetal/data/entity/response/BasePagedResponseEntity.kt
package com.github.gmetal.data.entity.response

import com.github.gmetal.data.entity.BaseMoviesDbEntity
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
open class BasePagedResponseEntity : BaseMoviesDbEntity() {

    @SerialName("page")
    var page: Int = 0

    @SerialName("total_results")
    var totalResults: Int = 0

    @SerialName("total_pages")
    var totalPages: Int = 0
}
